pandapipes/control/run_control.py

```diff
@@ -35,10 +35,19 @@ def prepare_run_ctrl(net, ctrl_variables):
     :return: ctrl_variables
     :rtype: dict
     """
+    ctrl_var = ctrl_variables
+
     if ctrl_variables is None:
         ctrl_variables  = prepare_run_control_pandapower(net, None)
         ctrl_variables["run"] = ppipe.pipeflow
 
+    if ('continue_on_divergence') in kwargs and (ctrl_var is None):
```

Comment:
Why is this `and (ctrl_var is None)`? I thought a case with some ctrl_var (but not continue_on_divergence) and continue_on_divergence in the kwargs could also happen, couldn't it?
I'd expect something like `and ((ctrl_var is None) or (continue_on_divergence not in ctrl_var.keys()))`

Reply:
Following cases I was thinking about:
- a user defines ctrl_var and continue_on_divergence, then always the ctrl_var is preferred.
- Only if ctrl_var is None, then look for a continue_on_divergence in kwargs
- in the default prepare_run_control_pandapower by default continue_on_divergence is set to False
Actually, what I see now is that you don't need this lines but you can just pass the kwargs. So I would change that if you agree.

In your case you say that the user doesn't need to define continue_on_divergence. I wouldn't do that as, if the user wants to define his/her own ctrl_variables, he should be aware of what it is doing.

Reply:
Maybe something like:
if ('continue_on_divergence') in kwargs and (ctrl_var is None or not 'continue_on_divergence'  in ctrl_var.keys())
following use cases: 
1. user defines ctrl_variables and no divergence, it takes divergence from ctrl_variables (user needs to define it)
2. user defines ctrl_variables (without divergence) and divergence, divergence is taking from the kwargs
3. user defines both, ctrl_varibales is preferred
4. user defines no ctrl_variables but divergence, divergence is used.

Same solution could be selected in case of check_each_level. What do you think?

Reply:
The suggestion from the last post sounds good to me. I am not sure though, whether in Case 3 (user defines both) ctrl_variables should be preferred, as it is slightly more indirect.
In the end, I think the main point is simply to have the behavior documented somewhere.

Reply:
Actually I think if a user takes the effort and defines ctrl_variables himself/herself, I don't think it is implicit especially if you think about the following case: In a multinet you can define ctrl_variables for each net. So you define divergence for each net and think about what happens if powerflow is not converging. 
So what would happen if gloabl strikes net specific: the globally defined divergence would overwrite all the ctrl_variables net specific divergence entries. I implemented it that way that it would only change divergence if there is none. If it is defined by the user for a net specifically, it won't change the entry. Therefore, I would like stick to this solution. 
But we could open a new issue if you like and discuss it in a bigger plenum.

Reply:
Okay, I see your point. Sounds good to me.